test/AlarmClockTest.cpp

```diff
@@ -0,0 +1,198 @@
+   EXPECT_TRUE(msToMicro <= totalTime) << "AlarmClock didn't sleep for long enough. Slept for: " << totalTime << " us, should be longer than " << msToMicro;
+   auto maxTime = msToMicro + GetTimingLeeway(milliseconds(ms));
+   EXPECT_TRUE(totalTime <= maxTime) << "AlarmClock took too long to expire. Took " << totalTime << " us. Should be less than " << maxTime;
+   std::cout << "Timeout was set for " << msToMicro << " us. Actually slept for " << totalTime << " us. Max timeout: " << maxTime << std::endl;
+}
+
+TEST_F(AlarmClockTest, oneSecondInMilliseconds) {
+   unsigned int ms = 1000;
+   StopWatch testTimer;
+   AlarmClock<milliseconds> alerter(ms);
+   WaitForAlarmClockToExpire(alerter);
+   auto totalTime = testTimer.ElapsedUs();
+   auto msToMicro = ConvertToMicroSeconds(milliseconds(ms));
+   EXPECT_TRUE(msToMicro <= totalTime) << "AlarmClock didn't sleep for long enough. Slept for: " << totalTime << " us, should be longer than " << msToMicro;
+   auto maxTime = msToMicro + GetTimingLeeway(milliseconds(ms));
+   EXPECT_TRUE(totalTime <= maxTime) << "AlarmClock took too long to expire. Took " << totalTime << " us. Should be less than " << maxTime;
+   std::cout << "Timeout was set for " << msToMicro << " us. Actually slept for " << totalTime << " us. Max timeout: " << maxTime << std::endl;
+}
+
+TEST_F(AlarmClockTest, millisecondsNotDivisibleBy500) {
+   unsigned int ms = 1300;
+   StopWatch testTimer;
+   AlarmClock<milliseconds> alerter(ms);
+   WaitForAlarmClockToExpire(alerter);
+   auto totalTime = testTimer.ElapsedUs();
+   auto msToMicro = ConvertToMicroSeconds(milliseconds(ms));
+   EXPECT_TRUE(msToMicro <= totalTime) << "AlarmClock didn't sleep for long enough. Slept for: " << totalTime << " us, should be longer than " << msToMicro;
+   auto maxTime = msToMicro + GetTimingLeeway(milliseconds(ms));
+   EXPECT_TRUE(totalTime <= maxTime) << "AlarmClock took too long to expire. Took " << totalTime << " us. Should be less than " << maxTime;
+   std::cout << "Timeout was set for " << msToMicro << " us. Actually slept for " << totalTime << " us. Max timeout: " << maxTime << std::endl;
+}
+
+TEST_F(AlarmClockTest, secondsSimple) {
+   int sec = 1;
+   StopWatch testTimer;
+   AlarmClock<seconds> alerter(sec);
+   WaitForAlarmClockToExpire(alerter);
+   auto totalTime = testTimer.ElapsedUs();
+   auto secToMicro = ConvertToMicroSeconds(seconds(sec));
+   EXPECT_TRUE(secToMicro <= totalTime) << "AlarmClock didn't sleep for long enough. Slept for: " << totalTime << " sec, should be longer than " << secToMicro;
+   auto maxTime = secToMicro + (kTimingLeeway * sec);
+   EXPECT_TRUE(totalTime <= maxTime) << "AlarmClock took too long to expire. Took " << totalTime << " sec. Should be less than " << maxTime;
+   std::cout << "Timeout was set for " << secToMicro << " us. Actually slept for " << totalTime << " us. Max timeout: " << maxTime << std::endl;
+}
+
+TEST_F(AlarmClockTest, AlarmClockVSStopWatch) {
```

Comment:
Remove this test or make it significantly shorter for the actual checked in code